docs: add release notes for 0.24.0 (#7680)
determined-ai/determined: docs/release-notes.rst

 Release Notes
###############

**************
 Version 0.24
**************

Version 0.24.0
==============

**Release Date:** August 18, 2023

**Breaking Changes**

> -  API: Remove PyTorchLightningAdapter, which was deprecated in 0.23.1. We recommend that PyTorch

Reviewer: ```suggestion
-  API: Remove ``LightningAdapter``, which was deprecated in 0.23.1 (June 2023). We recommend that PyTorch
```

   Lightning users migrate to the :ref:`Core API <core-getting-started>`.

**New Features**

> -  Environments: Add the following PyTorch2.0 images (experimental): ``pytorch==2.0.1``,
>    ``python==3.10.12``, and ``CUDA==11.8``.

Reviewer: ```suggestion
-  Environments: Add experimental PyTorch 2.0 images containing PyTorch 2.0.1, Python 3.10.12, and
   (for the GPU image) CUDA 11.8.
```

**Bug Fixes**

> -  Users: Fix an issue that caused the CLI command ``det user list`` to always return false for the

Reviewer: ```suggestion
-  Users: Fix an issue that caused the CLI command ``det user list`` to always show "false" in the
```

>    remote column.

Reviewer: ```suggestion
   "remote" column.
```
